### PR TITLE
feature(admin-api): push version

### PR DESCRIPTION
**Pull request by JYisus**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [ ] I have created tests for my code changes, and the tests are passing.
- [ ] I have executed the pre-commit hooks locally.
- [ ] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by kafkaphoenix — DISMISSED** [review]

File: engine/admin-api/adapter/repository/mongodb/versionrepository/repository.go [hunk 1]
```diff
@@ -104,20 +101,20 @@ func (r *VersionRepoMongoDB) GetByTag(ctx context.Context, productID, tag string
...
 	if errors.Is(err, mongo.ErrNoDocuments) {
-		return nil, apperrors.ErrVersionNotFound
+		return nil, version.ErrVersionNotFound
 	}
 
 	return mapDTOToEntity(versionDTO), err
 }
 
-func (r *VersionRepoMongoDB) Update(productID string, version *entity.Version) error {
+func (r *VersionRepoMongoDB) Update(productID string, updatedVersion *entity.Version) error {
```
Comment (alexrodfe) [inline engine/admin-api/adapter/repository/mongodb/versionrepository/repository.go:110]: imo I'd have this "updatedVersion" as versionToUpdate or something similar that tells this is the version that is going to be updated, not that it has been updated

File: engine/admin-api/domain/usecase/version/create_test.go [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+	"github.com/stretchr/testify/suite"
+)
+
+type createVersionSuite struct {
+	suite.Suite
+
+	logger      logr.Logger
+	versionRepo *mocks.MockVersionRepo
+	productRepo *mocks.MockProductRepo
+	//versionService   *mocks.MockVersionService
```
Comment (alexrodfe) [inline engine/admin-api/domain/usecase/version/create_test.go:28]: watch out unused variable

File: engine/admin-api/domain/usecase/version/create_test.go [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+			{
+				Key:   "keyA",
+				Value: "value1",
+			},
+		},
+		Workflows: []entity.Workflow{
+			{
+				Name: "go-classificator",
+				Type: entity.WorkflowTypeData,
+				//Stream:    "",
```
Comment (alexrodfe) [inline engine/admin-api/domain/usecase/version/create_test.go:197]: ditto

File: engine/admin-api/domain/usecase/version/handler_test.go [hunk 4]
```diff
@@ -2,275 +2,275 @@
 
 package version_test
 
-import (
```
Comment (alexrodfe) [inline engine/admin-api/domain/usecase/version/handler_test.go:5]: why is everything commented?

Comment (alexrodfe) [reply engine/admin-api/domain/usecase/version/handler_test.go:5]: this file is deleted in the start version pr btw, in favor to a new version test suite

File: engine/admin-api/internal/errors/errors.go [hunk 5]
```diff
@@ -27,33 +20,6 @@ var (
 	// ErrCreatingDashboard error.
 	ErrCreatingDashboard = errors.New("error creating dashboard")
```
Comment (alexrodfe) [inline engine/admin-api/internal/errors/errors.go:21]: I believe these two last errors are no longer needed

**Review by alexrodfe — CHANGES_REQUESTED** [review]
Great, look good
I got a couple of questions

File: engine/admin-api/adapter/repository/mongodb/versionrepository/repository.go (hunk 1 above)
Comment (JYisus) [reply engine/admin-api/adapter/repository/mongodb/versionrepository/repository.go:110]: Well, it is the updated version indeed, but not persisted. I mean, the version already has the changes.

File: engine/admin-api/domain/usecase/version/handler_test.go (hunk 4 above)
Comment (JYisus) [reply engine/admin-api/domain/usecase/version/handler_test.go:5]: Yep, I was using it as reference, but I forgot to delete it

**Review by kafkaphoenix — APPROVED** [review]

**Review by kafkaphoenix — APPROVED** [review]

**Review by alexrodfe — APPROVED** [review]

**Pull request merged**